argusAI/event_similarity_processor_v2.py

```diff
@@ -0,0 +1,234 @@
+SLEEP_INTERVAL = 1  # Sleep for 1 second between processing cycles
+
+
+class BgeSmallEnEmbeddingModel(ONNXMiniLM_L6_V2):
+    """
+    Compact English text embedding model by BAAI, released Sep 2023.
+    Trained on large-scale paired data with RetroMAE and contrastive learning.
+    Produces high-quality 384-dim embeddings, optimized for efficiency in semantic search, classification, and clustering.
+    """
+
+    MODEL_NAME: str = "bge-small-en-v1.5"
+    DOWNLOAD_PATH: Path = Path.home() / ".cache" / "chroma" / "onnx_models" / MODEL_NAME
+    EXTRACTED_FOLDER_NAME: str = "onnx"
+    ARCHIVE_FILENAME: str = "onnx.tar.gz"
+    MODEL_DOWNLOAD_URL: str = (
+        "https://scylla-qa-public.s3.us-east-1.amazonaws.com/ArgusAI/bge-small-en-v1.5/onnx.tar.gz"
+    )
+    _MODEL_SHA256: str = "e7d1743b0c08f55c687cff6af696683398682f9ab4fb3cad1be644ee5553a72d"
+
+
+class EventSimilarityProcessorV2:
+    """
+    Processes unprocessed SCT events by generating embeddings and storing them in severity-specific tables.
+    """
+
+    def __init__(self, stop_event: Event | None = None) -> None:
+        """
+        Initialize the processor with embedding model and sanitizer.
+
+        Args:
+            stop_event: Optional threading.Event to signal shutdown
+        """
+        self.embedding_model = BgeSmallEnEmbeddingModel()
+        self.sanitizer = MessageSanitizer()
+        self.stop_event = stop_event or Event()
+        self.db = ScyllaConnection()
+        self.processed_count = 0
+        self.error_count = 0
+        LOGGER.info("EventSimilarityProcessorV2 initialized")
+
+    def process_unprocessed_events(self) -> None:
+        """
+        Main processing loop that continuously reads and processes unprocessed events.
+        """
+        LOGGER.info("Starting event processing loop")
+
+        while not self.stop_event.is_set():
+            try:
+                batch_processed = self._process_batch()
+                if batch_processed == 0:
+                    # No events to process, sleep before next iteration
+                    time.sleep(SLEEP_INTERVAL)
+                else:
+                    LOGGER.info(f"Processed {batch_processed} events in this batch")
+            except Exception as e:
+                LOGGER.error(f"Error in processing loop: {e}", exc_info=True)
+                self.error_count += 1
+                time.sleep(SLEEP_INTERVAL)
+
+        LOGGER.info(f"Processing loop stopped. Total processed: {self.processed_count}, Errors: {self.error_count}")
+
+    def _process_batch(self, batch_size: int = 100) -> int:
+        """
+        Process a batch of unprocessed events.
+
+        Args:
+            batch_size: Maximum number of events to process in one batch
+
+        Returns:
+            Number of events processed in this batch
+        """
+        # Fetch unprocessed events using raw query
+        query = f"SELECT run_id, severity, ts FROM sct_unprocessed_events LIMIT {batch_size}"
```

Comment:
Could use a custom `table_name()` class method instead of hardcoded table name (You will need to define a table name on the model as well).